File: GetWhoISParsed.vb

Imports System.Web.Script.Serialization
Public Class jsonwhoisapi

Public Function GetWhoISParsed(ByVal vDomain As String) As WhoISParsed
      
      Try
           Dim strAccountNumber As String = "" 'Set your Account Number
           Dim strAPIKey As String = ""  'Set Your API Key
           
           Dim request = TryCast(System.Net.WebRequest.Create("https://jsonwhoisapi.com/api/v1/whois?identifier=" & HttpUtility.UrlEncode(vDomain)), System.Net.HttpWebRequest)

            request.Method = "GET"
            request.ContentType = "application/json"
            
            Dim svcCredentials As String = Convert.ToBase64String(ASCIIEncoding.ASCII.GetBytes(strAccountNumber & ":" & strAPIKey))

           request.Headers.Add("authorization", "Basic " & svcCredentials)

            Dim strResponseContent As String
            Using response = TryCast(request.GetResponse(), System.Net.HttpWebResponse)
                Using reader = New System.IO.StreamReader(response.GetResponseStream())
                    strResponseContent = reader.ReadToEnd()
                End Using
            End Using

            Dim serializer As New JavaScriptSerializer
            Dim oWhoIsParsed As WhoISParsed
            oWhoIsParsed = serializer.Deserialize(Of WhoISParsed)(strResponseContent)
            Return oWhoIsParsed.created

        Catch ex As Exception
            'Handle any errors
        End Try

    End Function
End Class


Public Class Owner
        Public Property handle As Object
        Public Property type As Object
        Public Property name As String
        Public Property organization As String
        Public Property email As String
        Public Property address As String
        Public Property zipcode As String
        Public Property city As String
        Public Property state As String
        Public Property country As String
        Public Property phone As String
        Public Property fax As String
        Public Property created As Object
        Public Property changed As Object
    End Class

    Public Class Admin
        Public Property handle As Object
        Public Property type As Object
        Public Property name As String
        Public Property organization As String
        Public Property email As String
        Public Property address As String
        Public Property zipcode As String
        Public Property city As String
        Public Property state As String
        Public Property country As String
        Public Property phone As String
        Public Property fax As String
        Public Property created As Object
        Public Property changed As Object
    End Class

    Public Class Tech
        Public Property handle As Object
        Public Property type As Object
        Public Property name As String
        Public Property organization As String
        Public Property email As String
        Public Property address As String
        Public Property zipcode As String
        Public Property city As String
        Public Property state As String
        Public Property country As String
        Public Property phone As String
        Public Property fax As String
        Public Property created As Object
        Public Property changed As Object
    End Class

    Public Class Contacts
        Public Property owner As Owner()
        Public Property admin As Admin()
        Public Property tech As Tech()
    End Class

    Public Class Registrar
        Public Property id As String
        Public Property name As Object
        Public Property email As Object
        Public Property url As Object
        Public Property phone As Object
        Public Property table As Table
    End Class
    
    Public Class Table
    End Class
    
    Public Class WhoISParsed
        Public Property name As String
        Public Property created As Object
        Public Property changed As Object
        Public Property expires As Object
        Public Property dnssec As Object
        Public Property registered As Object
        Public Property status As Object
        Public Property nameservers As Object()
        Public Property contacts As Contacts
        Public Property registrar As Registrar
        Public Property throttled As Object
    End Class

File: GetWhoISParsed.cs
using System.Web.Script.Serialization;

public class jsonwhoisapi
{
    public WhoISParsed GetWhoISParsed(string vDomain)
    {
        try
        {
            string strAccountNumber = "";
            string strAPIKey = "";
            var request = System.Net.WebRequest.Create("https://jsonwhoisapi.com/api/v1/whois?identifier=" + HttpUtility.UrlEncode(vDomain)) as System.Net.HttpWebRequest;
            request.Method = "GET";
            request.ContentType = "application/json";
            string svcCredentials = Convert.ToBase64String(ASCIIEncoding.ASCII.GetBytes(strAccountNumber + ":" + strAPIKey));
            request.Headers.Add("authorization", "Basic " + svcCredentials);
            string strResponseContent;
            using (var response = request.GetResponse() as System.Net.HttpWebResponse)
            {
                using (var reader = new System.IO.StreamReader(response.GetResponseStream()))
                {
                    strResponseContent = reader.ReadToEnd();
                }
            }

            JavaScriptSerializer serializer = new JavaScriptSerializer;
            WhoISParsed oWhoIsParsed;
            oWhoIsParsed = serializer.Deserialize<WhoISParsed>(strResponseContent);
            return oWhoIsParsed.created;
        }
        catch (Exception ex)
        {
        }
    }
}


public class Table
    {
    }

    public class Registrar
    {
        public Table table { get; set; }
        public string id { get; set; }
        public object name { get; set; }
        public object email { get; set; }
        public object url { get; set; }
        public object phone { get; set; }
     }
    
    Public Class Contacts
    {
        Public Owner owner { get; set; }
        Public Admin admin { get; set; }
        Public Tech tech { get; set; }
    }


    public class WhoISParsed
    {
        public string name { get; set; }
        public object created { get; set; }
        public object changed { get; set; }
        public object expires { get; set; }
        public object dnssec { get; set; }
        public object registered { get; set; }
        public object status { get; set; }
        public IList<object> nameservers { get; set; }
        public object contacts { get; set; }
        public Registrar registrar { get; set; }
        public object throttled { get; set; }
    }
    
    Public Class Owner
    {
       public object handle { get; set; }
       public object type { get; set; }
       public string name { get; set; }
       public string organization { get; set; }
       public string email { get; set; }
       public string address { get; set; }
       public string zipcode { get; set; }
       public string city { get; set; }
       public string state { get; set; }
       public string country { get; set; }
       public string phone { get; set; }
       public string fax { get; set; }
       public object created { get; set; }
       public object changed { get; set; }
 }

    Public Class Admin
    {
       public object handle { get; set; }
       public object type { get; set; }
       public string name { get; set; }
       public string organization { get; set; }
       public string email { get; set; }
       public string address { get; set; }
       public string zipcode { get; set; }
       public string city { get; set; }
       public string state { get; set; }
       public string country { get; set; }
       public string phone { get; set; }
       public string fax { get; set; }
       public object created { get; set; }
       public object changed { get; set; }
   }

    Public Class Tech
    {
       public object handle { get; set; }
       public object type { get; set; }
       public string name { get; set; }
       public string organization { get; set; }
       public string email { get; set; }
       public string address { get; set; }
       public string zipcode { get; set; }
       public string city { get; set; }
       public string state { get; set; }
       public string country { get; set; }
       public string phone { get; set; }
       public string fax { get; set; }
       public object created { get; set; }
       public object changed { get; set; }
   }


    
